In [18]:
import numpy as np

Para el numerador usamos cuadratura de Gauss-Legendre debido a que la integral esta entre limites a y b.

Los limites los encontramos usando c = λν para encontrar los valores de v inicial y v final
luego reemplazamos en hν/kBT esto nos dio:
a = 6.239
b = 24.955


In [19]:
n = 20
a = 6.239
b = 24.955
f_nume = lambda x: (x**3)/(np.exp(x)-1)
Roots_nume, Weights_nume = np.polynomial.legendre.leggauss(n)

t = 0.5*( (b-a)*Roots_nume + a + b )
numerador = 0.5*(b-a)*np.sum(Weights_nume*f_nume(t))
numerador

0.7872948073429804

Para el denominador usamos cuadratura de Gauss-Laguerre debido a que la integral esta definida entre 0 e inf.

Para usas esta cuadratura debemos tener la integral de la forma e^x*f(x) por lo tanto cambiamos la funcion llegando a e^(-x)(x**3/(1-e^-x))

In [20]:
Roots_deno, Weights_deno= np.polynomial.laguerre.laggauss(n)
f_deno = lambda x: (x**3)/(1-np.exp(-x))

denominador = np.sum(Weights_deno*f_deno(Roots_deno))
denominador

6.493939402219589

In [21]:
f = numerador/denominador
f

0.12123531782170431

Según la página del IDEAM Bogotá-UV el porcentaje de ultravioleta es 7.2%. ¿A
qué se debe esta diferencia?

El 12% obtenido en los puntos anteriores representa la cantidad de rayos UV emitidos por el sol, sin embargo, el valor medido en la tierra varia a 7.2% debido a la dispersion de la luz causada por las moleculas en la atmosfera haciendo a su vez que podamos ver el cielo de color azul debido a que tiene una longitud de onda mas corta. Este fenomeno se conoce como dispersion de Rayleigh indicando que cuando las moleculas tienen un tamaño menor o similar al de la longitud de onda, la luz sufrira una dispersion resultado de la polarizacion electrica.